frontend/src/components/organisms/ProfileForm.tsx

```diff
@@ -40,8 +50,102 @@ interface ProfileFormProps {
   userDetails: formData;
 }
 
+interface ModalBodyProps {
+  userDetails: formData;
+  handleClose: () => void;
+}
+
+/** Confirmation Modal To Delete An Account */
+const ModalBody = ({ userDetails, handleClose }: ModalBodyProps) => {
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<DeleteAccountFormValues>();
+
+  //changeUserStatus handles deleting user from prisma and firebase
+  const { mutateAsync: changeUserStatus, error: changeUserStatusError } =
+    useMutation({
+      mutationFn: async () => {
+        const { data } = await api.delete(`/users/${userDetails.id}`);
+        return data;
+      },
+      retry: false,
+      onSuccess: () => {
+        console.log("DELETED");
+      },
+    });
+
+  //Variables for signing out user
+  const { error, signOutUser } = useAuth();
+  const queryClient = useQueryClient();
+  const router = useRouter();
+
+  //Signs out user and redirects them to login page
+  const handleSignOut = async () => {
+    try {
+      await signOutUser();
+      queryClient.clear(); // Clear cache for react query
+      router.replace("/login");
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  /**  When delete form is submitted, we will delete user, end their session,
+   * and redirect them to login page
+   */
+  const handleDeleteAccount = async () => {
+    await changeUserStatus();
+    await handleSignOut();
```

Comment:
Wrap this in a try catch, it is possible that it fails and the user will need some feedback that the action failed